frontend/app/src/pages/auth/hooks/use-cloud.ts

```diff
@@ -0,0 +1,74 @@
+import { cloudApi } from '@/lib/api/api';
+import { APICloudMetadata } from '@/lib/api/generated/cloud/data-contracts';
+import { useApiError } from '@/lib/hooks';
+import { useQuery } from '@tanstack/react-query';
+import { AxiosError } from 'axios';
+import { useCallback, useMemo } from 'react';
+
+export default function useCloud(tenantId?: string) {
```

Comment:
IMO this should be a provider, so we can fetch this once and just have a boolean flag everywhere that we can check. I think it'd be amazing to consolidate the tenant, organization, and this into a single provider if we can, but that's probably work for later